### PR TITLE
#3 implemented basic architecture

### PR DESCRIPTION
**Pull request by MGA-dotSource**
all basically required interfaces for the application modules created
added some example test cases

### CODE REVIEW

**Comment by lauraluiz** [comment]
Put also @schleichardt as assignee or mention us, any way works :)

**Comment by MGA-dotSource** [comment]
ok

File: common/src/main/java/com/commercetools/sunrise/payment/service/PaymentAdapterService.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    CompletionStage<List<PaymentMethodInfo>> findAvailablePaymentMethods(); // TODO: add parameters possibly required
+
+    /**
+     * Creates a new payment object at the CTP and theirby starts a new payment transaction workflow.
+     * Possibly existing payments will be cancelled but not deleted.
+     * @param methodID the ID of the selected payment method
+     * @param data the wrapper object for all possible needed data
+     */
+    void createPayment(String methodID, CreatePaymentDataProvider data);
+
```
Comment (butenkor) [inline common/src/main/java/com/commercetools/sunrise/payment/service/PaymentAdapterService.java]: how about adding ```add/create Transaction```: https://github.com/commercetools/commercetools-sunrise-java-payment/issues/3#issuecomment-233384680

Comment (butenkor) [inline common/src/main/java/com/commercetools/sunrise/payment/service/PaymentAdapterService.java]: I would rather go for naming ```setPayment``` and ```addTransaction```, wdyt?

**Comment by MGA-dotSource** [comment]
as all changes will be included in another pull request I'll close this one

**Pull request closed without merging**